spec/factories.rb

```diff
@@ -4,9 +4,24 @@
   factory :book do
     genre { Faker::HitchhikersGuideToTheGalaxy.location }
     author { Faker::Name.name }
-    image { Faker::String.random }
+    image { Faker::Internet.url }
     title { Faker::LordOfTheRings.location }
     publisher { Faker::Name.name }
     year { Faker::Number.number(4) }
   end
+
+  factory :user do
```

Comment:
It not a good pratice to put all factories in the same file. Better create a repository `factories/`, and in this one have a file by model. `users.rb` `books.rb` and `rents.rb`

Reply:
You did not separate factories in multiple files.